### ISSUE TITLE
[Bug]: `aws_berockagent_agent` does not return `agent_version`

### ISSUE THREAD

**Issue by kmcduffee-verisk** [hashicorp/terraform-provider-aws#37321]
### Terraform Core Version

1.8.2

### AWS Provider Version

5.48.0

### Affected Resource(s)

* aws_bedrockagent_agent
* aws_bedrockagent_agent_alias

### Expected Behavior

When creating or updating the agent, `agent_version` should be included in the resource attributes. This would then allow for updating an associated `aws_bedrockagent_agent_agent_alias` to point to the new version.

### Actual Behavior

When creating or updating an `aws_bedrockagent_agent` resource, the `agent_version` is not returned. This makes it difficult to make changes to an associated `aws_bedrockagent_agent_alias` resource, as it's not possible to set its `routing_configuration` block to point to the new version.

### Terraform Configuration Files

```
resource "aws_bedrockagent_agent" "demo" {
  agent_name                  = "demo"
  agent_resource_role_arn     = aws_iam_role.demo.arn
  foundation_model            = "anthropic.claude-3-sonnet-20240229-v1:0"
  idle_session_ttl_in_seconds = 60
  instruction = "Just testing."
}
  
output "agent_version" {
  value = aws_bedrockagent_agent.demo.agent_version
}
  ```

### Steps to Reproduce

1. Apply included terraform.
2. Note that `agent_version` is not returned.

### Would you like to implement a fix?

None

**Comment by kmcduffee-verisk** [comment]
Currently I am solving the alias issue with the following:

```
resource "aws_bedrockagent_agent_alias" "demo" {
  agent_alias_name = "demo"
  agent_id         = aws_bedrockagent_agent.demo.agent_id
  description      = "Demo"
  routing_configuration {
    agent_version = data.external.latest_agent_version.result.agent_version
  }
}

data "external" "latest_agent_version" {
  program = ["bash","${path.module}/scripts/get_agent_details.sh",aws_bedrockagent_agent.demo.agent_id]
  depends_on = [ aws_bedrockagent_agent.demo ]
}

```
#!/usr/bin/env bash

AGENT_ID=$1
AGENT_VERSION=$(aws bedrock-agent list-agents --query "agentSummaries[?agentId=='${AGENT_ID}'].latestAgentVersion" --output text)

jq -n --arg agent_version "${AGENT_VERSION}" '{"agent_version":$agent_version}'
```

**Comment by acwwat** [comment]
Just adding some observations here since I've been experimenting with the Bedrock Agents resources recently.

It seems that the `agent_version` in the context of `aws_bedrockagent_agent` is not very helpful because the value is supposedly hardcoded to `DRAFT` if you look at the [API reference](https://docs.aws.amazon.com/bedrock/latest/APIReference/API_agent_Agent.html#bedrock-Type-agent_Agent-agentVersion). But the actual behavior is also different or changing. I can see that the field is defined in the `aws_bedrockagent_agent` resource code as a computed attribute but it's being set to null because the AWS API is not returning it! When I called `aws bedrock-agent get-agent` in the CLI, the response does NOT include the `agentVersion` field.

It also doesn't help when versions are created only when an alias is created. So what I think we need is a new data source `aws_bedrockagent_agent_versions` to fetch the available versions for creating an alias pointing to a specific version.

**Comment by drewtul** [comment]
Per @acwwat The version for a agent created via `aws_bedrockagent_agent` is always `DRAFT`, using `aws_bedrockagent_agent_alias` without `routing_configuration` will create a new version of the agent from the currently prepared `DRAFT`. That appears to be the only way to create a new version at this time.

You can also using the data source in #38792 to look up specific versions if you want to create an `aws_bedrockagent_agent_alias` to a specific version that already exists.

We should probably deprecate and remove `agent_version` as the API isn't returning it and it could only be `DRAFT` it's not providing anything useful.

As it stands I don't think this is a bug we can fix, the `CreateAgentAlias` API call is the only call that creates new versions of the agent.

**Comment by gmilowamw** [comment]
Unfortunately there is no way to update an existing `aws_bedrockagent_agent_alias` and still have it create a new version, even without `routing_configuration`. The only way I can get a new version created, using Terraform, is to create a whole new `aws_bedrockagent_agent_alias`. Feels like this is still an unresolved issue.

**Comment by drewtul** [comment]
Correct, there is no API to create a new version of an agent outside of `CreateAlias`, if there was we could create a version when the agent is modified and expose the latest version to allow the alias to be pointed to it.

**Comment by v-yarotsky** [comment]
While not ideal, I think I've found a workaround that can be used to create a new version of the agent whenever there's a change, together with updating an `aws_bedrockagent_agent_alias`.

Something along the lines of:
```hcl
# This is an ephemeral alias that gets recreated whenever the agent is changed
resource "aws_bedrockagent_agent_alias" "ephemeral" {
  agent_id         = aws_bedrockagent_agent.agent.id
  agent_alias_name = "myagent-ephemeral"

  description = "Ephemeral alias used as a hack to trigger new agent version creation."

  lifecycle {
    replace_triggered_by = [
      aws_bedrockagent_agent.agent
    ]
  }
}

# This is an alias with a stable ID that is updated to point at a new version whenever the agent is updated
resource "aws_bedrockagent_agent_alias" "main" {
  agent_id         = aws_bedrockagent_agent.agent.id
  agent_alias_name = "myagent"

  routing_configuration {
    agent_version = aws_bedrockagent_agent_alias.ephemeral.routing_configuration[0].agent_version
  }
}

```

**Comment by drewtul** [comment]
With the latest release you can cause an existing alias to create a new version by updating the description. I also exposed prepared_at on the agent resource so you can use that to know when an agent has a configuration change to create the new version.

**Comment by mattgillard** [comment]
> While not ideal, I think I've found a workaround that can be used to create a new version of the agent whenever there's a change, together with updating an `aws_bedrockagent_agent_alias`.
> 

Thank you!

This comment led me here and I implemented something similar with SST (which uses Pulumi behind the scenes which uses the AWS terraform provider :) )
There is an extra level of difficulty though replace_triggered_by of another resource is not available so I had to make a hack with a hash of some params that may change in the agent which should trigger the new alias version which gets stored in an alias tag. So if any of them change the alias will get regenerated. Should also add function code but this was just a poc.

```typescript
    const agentConfigSignature = $util.all([
      agent.agentName,
      agent.description,
      agent.foundationModel,
      agent.agentResourceRoleArn,
      agent.instruction,
    ]).apply(([name, description, model, roleArn, instruction]) => {
      const hash = crypto.createHash("sha256");
      // Create a consistent string representation of the properties to hash
      const configToHash = {
        name,
        description,
        model,
        roleArn,
        instruction,
      };
      hash.update(JSON.stringify(configToHash));
      return hash.digest("hex");
    });

    const agentAliasEphemeral = new aws.bedrock.AgentAgentAlias("AgentAliasEphemeral", {
      agentAliasName: "agentAliasEphemeral",
      agentId: agent.id,
      description: "AliasEphemeral",
      tags: { 
        "agentConfigSignature": agentConfigSignature, // Assign the computed signature
      }
    },
      {
        dependsOn: [agent, actionGroup, actionGroup2],
        replaceOnChanges: ["agentId", "tags"],
        deleteBeforeReplace: true,
      });

    const agentAlias = new aws.bedrock.AgentAgentAlias("agentAlias", {
      agentAliasName: "live",
      agentId: agent.id,
      description: "Agent Alias",
      //@ts-ignore bug with AgentAgentAlias types
      routingConfigurations: [{
        agentVersion: agentAliasEphemeral.routingConfigurations![0].agentVersion
      }]
    }, {
      dependsOn: [agentAliasEphemeral],
    });
```
